openlcsd/flow/tasks.py

```diff
@@ -753,12 +770,68 @@ def save_container_components(context, engine):
     engine.logger.info('Finished saving container components.')
 
 
-def get_component_source_path(context, engine):
-    raise NotImplementedError
+def get_components_product_from_source_container(context, engine):
+    engine.logger.info('Start to get components data from source container...')
+    config = context.get('config')
+    sc_nvr = context.get('package_nvr')
+    # Get components from the source conainer ifself.
+    srpm_dir = context.get('srpm_dir')
+    misc_dir = context.get('misc_dir')
+    sc_handler = SourceContainerHandler(config)
+    components = sc_handler.get_container_components(srpm_dir, misc_dir,
+                                                     sc_nvr)
+    engine.logger.info('Finished getting components from source container')
+    context['components'] = components
+
+
+def import_types_components(context, engine, nvr_list, src_dir, comp_type):
+    """
+    In a source container, it has different type of component. The different
+    components have different src_dir. This founction will be define the post
+    data for the differnt type component.
+    """
+    cli = context.get('client')
+    scan_options = {
+        'crypto_scan': context.get('crypto_scan'),
+        'copyright_scan': context.get('copyright_scan'),
+    }
+    url = '/sources/import/'
+    msg = 'Start to fork imports for {} components...'.format(len(nvr_list))
+    data = {
+            'src_dir': src_dir,
+            'package_nvrs': nvr_list
+    }
+    data.update(scan_options)
+    engine.logger.info(msg)
+    cli.post(url, data=data)
+    msg = '-- Forked import tasks for below source components:{}'.format(
+            "\n\t" + "\n\t".join(nvr_list))
+    engine.logger.info(msg)
 
 
-def fork_component_imports(context, engine):
-    raise NotImplementedError
+def fork_components_imports(context, engine):
+    """
+    Fork components tasks
+    """
+    srpm_nvr_list = []
+    misc_nvr_list = []
+    for component in context.get('components')['SRPM']:
+        srpm_nvr = "-".join((component.get('name'),
+                            component.get('version'),
+                            component.get('release')))
+        srpm_nvr_list.append(srpm_nvr)
+    for component in context.get('components')['UNKNOWN']:
+        misc_nvr = "-".join((component.get('name'),
+                            component.get('version'),
+                            component.get('release')))
```

Comment:
srpm_nvr and misc_nur use same code, can remove one

Reply:
Updated.